relayer/src/event/monitor.rs

```diff
@@ -336,32 +354,55 @@ impl EventMonitor {
 
             let result = rt.block_on(async {
                 tokio::select! {
-                    Some(batch) = batches.next() => Ok(batch),
+                    Some(batch) = batches.next() => batch,
                     Some(err) = self.rx_err.recv() => Err(Error::WebSocketDriver(err)),
                 }
             });
 
             match result {
                 Ok(batch) => self.process_batch(batch).unwrap_or_else(|e| {
-                    warn!(chain.id = %self.chain_id, "{}", e);
+                    error!(chain.id = %self.chain_id, "{}", e);
                 }),
+                Err(Error::SubscriptionCancelled(reason)) => {
+                    error!(chain.id = %self.chain_id, "subscription cancelled, reason: {}", reason);
+
+                    self.propagate_error(Error::SubscriptionCancelled(reason))
```

Comment:
As a minor suggestion: 

Would it improve things if we propagate the error (which triggers packet clearing) __after__ a successful `self.reconnect();`? I can imagine that reconnection could be unsuccessful, or may take some time, multiple tries, and then not sure about what happens with packet clearing.

Reply:
Good point, I was actually wondering about that at some point. I eventually decided to emit the error before the reconnection because it seemed to me that clearing the packets did not require the WebSocket connection to be live. Am I mistaken here?